### PR TITLE
Fix tar file being removed before extension install

### LINKED ISSUE #2688: All 3 new methods of installing extensions appear broken in 4.2.3
https://github.com/lensapp/lens/pull/2435 should have enabled installing extensions in 3 ways:

- `lens://app/extensions/install/@mirantis/lens-extension-cc` from a browser (this was https://github.com/lensapp/lens/issues/2422)

This fails:
image1

- Paste `https://www.npmjs.com/package/@mirantis/lens-extension-cc` in the "Install extension" box in Lens (this was  https://github.com/lensapp/lens/issues/1708)

This fails:
image2

- Paste `http://registry.npmjs.com/@mirantis/lens-extension-cc` in the "Install extension" box in Lens (this was https://github.com/lensapp/lens/issues/1707)

This fails:
image3

All 3 methods should work now, as of Lens 4.2.2, based on the PR and milestone.

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes #2688

### CODE REVIEW

File: src/renderer/components/+extensions/extensions.tsx [hunk 1]
```diff
@@ -172,8 +172,16 @@ async function createTempFilesAndValidate({ fileName, dataP }: InstallRequest, d
...
-
-  disposer.push(() => fse.unlink(tempFile));
+  let id: string | undefined = undefined;
+
+  disposer.push(() => {
+    // This is necessary so that this file is only removed on the error condition.
+    // if `id` is set then when the disposer is optionally called in `unpackExtension`
+    // this file will not be removed.
+    if (!id) {
+      fse.unlink(tempFile).catch(noop);
```
Comment (jakolehm) [inline src/renderer/components/+extensions/extensions.tsx]: Should we use OS temp dir for these?

Comment (Nokel81) [reply src/renderer/components/+extensions/extensions.tsx]: Probably, would be much easier to deal with...

Comment (Nokel81) [reply src/renderer/components/+extensions/extensions.tsx]: So we already do... shall I just remove this cleanup?

**Review by jakolehm — APPROVED** [review]

**Review by jim-docker — APPROVED** [review]

**Comment by Nokel81** [comment]
Will start 4.2.4 once the hotbar PR is merged.

**Comment by stefcameron** [comment]
Thank you!

**Pull request merged**